Makefile

```diff
@@ -113,19 +127,74 @@ oci-build:
+	@mkdir -p $(shell dirname $@) && touch $@
+
+# legacy-manifest-files acts as an intermediary target by depending on all
+# bundle/manifest files being copied to the manifests directory. The manifest
+# yaml files are handled by the target immediately following this, where they
+# simply cp source to target using a wildcard. Put simply, legacy-manifest-files
+# maps to every yaml under bundle/manifest remapped to it's legacy path e.g.
+#   'bundle/manifests/loki.openshift.io_lokistacks.yaml' -> 'manifests/5.1/loki.openshift.io_lokistacks.yaml'
+# where manifests/5.1/loki.openshift.io_lokistacks.yaml is the target.
+legacy-manifest-files: legacy-manifest-dir
+legacy-manifest-files: legacy-csv
+legacy-manifest-files: $(filter-out %.clusterserviceversion.yaml, $(patsubst bundle/manifests/%.yaml, manifests/$(CLUSTER_LOGGING_VERSION)/%.yaml, $(wildcard bundle/manifests/*.yaml)))
+manifests/$(CLUSTER_LOGGING_VERSION)/%.yaml: bundle/manifests/%.yaml
+	@cp -v $^ $@
+
+
+# The CSV does not work under the previous rule, because the
+# CLUSTER_LOGGING_VERSION is part of the target filename so it requires it's
+# own target
+legacy-csv: manifests/$(CLUSTER_LOGGING_VERSION)/loki-operator.v$(CLUSTER_LOGGING_VERSION).clusterserviceversion.yaml
+manifests/$(CLUSTER_LOGGING_VERSION)/loki-operator.v$(CLUSTER_LOGGING_VERSION).clusterserviceversion.yaml: bundle/manifests/loki-operator.clusterserviceversion.yaml
+	@cp -v $^ $@
+
 # Generate bundle manifests and metadata, then validate generated files.
 .PHONY: bundle
-bundle: manifests $(KUSTOMIZE) $(OPERATOR_SDK)
+bundle: manifests legacy-manifest-dir $(KUSTOMIZE) $(OPERATOR_SDK)
 	$(OPERATOR_SDK) generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 	$(OPERATOR_SDK) bundle validate ./bundle
+	$(MAKE) legacy-manifest-files
 
 # Build the bundle image.
 .PHONY: bundle-build
 bundle-build:
 	$(OCI_RUNTIME) build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
 
+# Build and push the bundle image to a container registry.
+.PHONY: olm-deploy-bundle
+olm-deploy-bundle: bundle bundle-build
+	$(MAKE) oci-push IMG=$(BUNDLE_IMG)
+
+# Build and push the operator image to a container registry.
+.PHONY: olm-deploy-operator
+olm-deploy-operator: oci-build oci-push
+
+# Deploy the operator bundle and the operator via OLM into
+# an OpenShift cluster selected via KUBECONFIG.
+.PHONY: olm-deploy
+ifeq ($(or $(findstring openshift-logging,$(IMG)),$(findstring openshift-logging,$(BUNDLE_IMG))),openshift-logging)
+olm-deploy:
+	$(error Set variable REGISTRY_ORG to use a custom container registry org account for local development)
+else
+olm-deploy: olm-deploy-bundle olm-deploy-operator $(OPERATOR_SDK)
+	oc create ns $(CLUSTER_LOGGING_NS)
+	oc label ns/$(CLUSTER_LOGGING_NS) openshift.io/cluster-monitoring=true --overwrite
+	oc project $(CLUSTER_LOGGING_NS)
```

Comment:
This assumes an OpenShift environment, which I don't think we want. You should prefer `kubectl` and fallback to `oc` the same way `OCI_RUNTIME` is set at the top of the Makefile. You'll also need to remove the `project` idiom that's specific to `oc` and use the namespace flag `(-n [namespace])` instead. I am running OLM in KiND.